app/models/badge_achievement.rb

```diff
@@ -55,6 +55,6 @@ def award_credits
   end
 
   def single_award_badge?
-    badge.allow_multiple_awards == false
+    badge&.allow_multiple_awards == false
```

Comment:
The test `it { is_expected.to belong_to(:badge) }` in spec/models/badge_achievement_spec.rb was failing without this as `badge = nil` here, which is odd as the badge was being created fine by the factory and could be printed.

I'm wondering if it's some odd sequencing in how the badge is assigned somehow.

This doesn't seem correct - but I'm a bit stuck for options.

Reply:
Wow, this sent me down a rabbit hole. It turns out that the test is correct to run this code first, because validation can happen at the database level, so it has yet to know if `badge` exists or not. This conditional check could've been checking anything that is unrelated to these models. So I'd say what you have here is fine